kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/ProxyConfigConfigMap.java

```diff
@@ -144,12 +145,10 @@ private static void addResolvedRefsConditions(VirtualKafkaClusterStatusFactory s
                 .filter(ResolutionResult.ClusterResolutionResult::isAnyDependencyUnresolved)
                 .forEach(clusterResolutionResult -> {
 
-                    Comparator<LocalRef<?>> comparator = Comparator.<LocalRef<?>, String> comparing(LocalRef::getGroup)
-                            .thenComparing(LocalRef::getKind)
-                            .thenComparing(LocalRef::getName);
+                    Comparator<UnresolvedReference> comparator = Comparator.<UnresolvedReference, LocalRef> comparing(UnresolvedReference::to);
 
-                    LocalRef<?> firstUnresolvedDependency = clusterResolutionResult.unresolvedDependencySet().stream()
-                            .sorted(comparator).findFirst()
+                    LocalRef<?> firstUnresolvedDependency = clusterResolutionResult.unresolvedReferences().unresolved().stream()
```

Comment:
this needs to change since the result can also have dependencies that are resolved, but have `ResolvedRefs:False` in the Conditions